README.md

```diff
@@ -15,3 +15,45 @@ Checkout the code and sync submodules:
 Specify the board you want to build for:
 
 	$ make BOARD=mimxrt1010_evk
+
+## Filesystem Structure
+
+The filesystem is structured as follows;
+
+```
+        tinyuf2
+           |
+           +----hw                    Hardware specific files
+           |    |
+           |    +---bsp                 Board specific files
+           |    |
+           |    +---chip                Chip specific (multiple boards)
+           |
+           +----lib                   External supporting libraries
+           |
+           +----src                   tinyUF2 specific code
+```
+
+A board is selected by means of the BOARD option on the makefile command line. This references a particular `hw/bsp/<BOARD>` directory which then links to a specific chip and variant in the `hw/chip/` directories.  A new board is added under `/hw/bsp/<BOARD>`. It may optionally need a new chip or variant under the `hw/chip` directories or, more normally, it might need new pin options in `hw/chip/<FAMILY>/<GROUP>/pinning_options.h`.
+
+## Adding a new Board
+
+You can add a new board by copying an existing one that is closest to it in the `hw/` directory with something like `cp -a hw/bsp/mimxrt1010_evk hw/bsp/newboard`. Edit `hw/bsp/newboard/board.mk` to reference the appropriate `hw/chip` directory and the `board_config.h` file to adjust settings for the new board.
+
+In general entries in `hw/bsp/newboard/board.mk` should reference existing pinning definitions in the relavent `hw/chip` directory rather than being created afresh.
+
+## Non-standard options
+
+In general non-standard options are frowned upon because it changes the user experience of tinyUF2. However, on occasion these may be needed for specific reasons. The supported options are specified in `hw/bsp/<BOARD>/board_config.h` for a specific board. Not defining any of these options means sensible defaults will be applied;
+
+* `USB_VID`: Vendor ID for the board. Default is 0x239A.
```

Comment:
We can add a "(courtesy of Adafruit Industries)" here :)

Reply:
Agreed, done.